reconcile/utils/oc.py

```diff
@@ -668,6 +682,252 @@ def _run_json(self, cmd, allow_not_found=False):
+        if len(kind_group) > 1:
+            apigroup_override = kind_group[1]
+            for gv in self.api_kind_version[kind]:
+                if apigroup_override in gv:
+                    group_version = gv
+                    break
+        return(kind, group_version)
+
+    # this function returns a kind:apigroup/version map for each kind on the
+    # cluster
+    def get_api_resources(self):
+        # this returns a prefix:apis map
+        api_prefix = self.client.resources.parse_api_groups(
+            request_resources=False, update=True)
+        kind_groupversion = {}
+        for prefix, apis in api_prefix.items():
+            # each api prefix consists of api:versions map
+            for apigroup, versions in apis.items():
+                if prefix == 'apis' and len(apigroup) == 0:
+                    # the apis group has an entry with an empty api, but
+                    # querying the apis group with a blank api produces an
+                    # error.  We skip that condition with this hack
+                    continue
+                # each version is a version:obj map, where obj contains if this
+                # api version is preferred and optionally a list of kinds that
+                # are part of that apigroup/version.
+                for version, obj in versions.items():
+                    try:
+                        client_resources = self.client.resources
+                        r = client_resources.get_resources_for_api_version(
+                            prefix, apigroup, version, True)
+                    except kubernetes.client.exceptions.ApiException:
+                        # there may be apigroups/versions that require elevated
+                        # permisions, so go to the next one
+                        next
+                    # resources are in a kind:ResourceList map, where a
+                    # ResourceList contains the api endpoint
+                    # (apigroup/version/puralkind)
+                    for kind in r.keys():
+                        if len(apigroup) == 0:
+                            gv = f'{version}'
+                        else:
+                            gv = f'{apigroup}/{version}'
+
+                        # add the kind and apigroup/version to the set of api
+                        # kinds
+                        kind_groupversion = self.add_group_kind(
+                            kind, kind_groupversion, gv, obj.preferred)
+        return kind_groupversion
+
+    def get_items(self, kind, **kwargs):
+        k, group_version = self._parse_kind(kind)
+        obj_client = self._get_obj_client(group_version=group_version, kind=k)
+
+        namespace = ''
+        if 'namespace' in kwargs:
+            namespace = kwargs['namespace']
+            # for cluster scoped integrations
+            # currently only openshift-clusterrolebindings
+            if namespace != 'cluster':
+                if not self.project_exists(namespace):
```

Comment:
weird, why do we return empty list if the project does not exist?

Reply:
This should be mimicking the existing get_items functionality.  I'm not sure why this is done, but if this is not the same return as the oc client then it should be fixed.  I think this is correct though.